Add docs for webhooks feature. (#5425)
wso2/docs-is: en/includes/guides/webhooks/understanding-webhooks.md

# Understanding webhooks

Webhooks enable your applications to receive instant notifications from {{product_name}}, allowing you to respond immediately to important identity-related events. Common use cases for {{product_name}} webhooks include:

- Automatically provision users or send welcome notifications upon successful user registration.
- Update external user directories or CRM systems whenever a user profile changes.
- Notify external services or security teams immediately upon password changes or resets.

> - Integrate with SIEM systems to instantly detect and respond to suspicious login attempts or failed authentications.

Author: well, here it should not be in brackets right. we are not referring as an abbreviation rather a noun

Using webhooks, you can seamlessly integrate external systems with {{product_name}}'s identity flows. When an event happens, {{product_name}} immediately sends HTTP callbacks to your configured webhook endpoints. {{product_name}} webhooks use the [WebSubHub](https://websubhub.com/) protocol for secure and reliable event delivery.

## How webhooks work

> When an identity-related event (for example user registration, login success, profile update) occurs within {{product_name}}, it automatically generates a notification event. {{product_name}} sends this event as an HTTP request to your configured webhook URL. The request contains detailed information encoded in a structured JSON payload.

Reviewer: ```suggestion
When an identity event occurs (e.g., user registration, login, or profile update), {{product_name}} generates a notification and sends it as an HTTP request to your configured webhook URL. The request includes detailed information in a structured JSON payload.
```

Author: well lint rejects e.g

Each payload includes context such as the event type, user information, the timestamp of the action, and metadata you can use to track or correlate events.

> Webhook notifications follow the [Security Event Token (SET) specification (RFC 8417)](https://datatracker.ietf.org/doc/html/rfc8417). This ensures standardization and clarity.

Reviewer: ```suggestion
Webhook notifications follow the [Security Event Token (SET) specification (RFC 8417)](https://datatracker.ietf.org/doc/html/rfc8417), ensuring standardization and clarity.
```

## Webhook event types

{{product_name}} supports webhooks for identity-related events categorized as follows:

- **Login events**
      - Login success
      - Login failure
- **Registration events**
      - Registration success
      - Registration failure
- **Credential events**

>       - Credential updates

Reviewer: Will these events be triggered for users' passkey and TOTP authenticator enrolments as well?

Author: In future yes

- **User Account Management events**
      - User profile updates
      - User account status changes (lock/unlock, enable/disable, delete)

> Each webhook event payload includes structured details compliant with the [SET](https://datatracker.ietf.org/doc/html/rfc8417) specification, containing issuer information, timestamp, unique identifiers, user and organization context, and event-specific metadata.

Reviewer: shall we move this to L18. seems the same kind of info is repeated here.

Author: I think it's better to keep this and remove L18.

Author: Fixed this
